### ISSUE TITLE
dexopts attribute isn't wired into dx correctly

### ISSUE THREAD

**Issue by nkoroste** [bazelbuild/bazel#9383]
### Description of the problem / feature request:

The `dexopts` attribute on `android_binary` seems to be ignored and not propagated to the underling `dx` toold

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

Attempt to build and `android_binary` target that requires extra `dexopts`. Example:

```
android_binary(
    name = "transition-sample",
    manifest = "src/main/AndroidManifest.xml",
    srcs = glob([
        'src/main/java/**/*.java',
    ]),
    deps = [
        "@maven//:androidx_appcompat_appcompat",
        "//platform/transition/lib:core",
    ],
    visibility = ["//visibility:public"],
    resource_files = glob(["src/main/res/**"]),
    custom_package = 'com.foo.android.transitionexample',
    manifest_values = {
        "minSdkVersion": "19",
        "targetSdkVersion" : "26"
    },
    dexopts = [
        "--force-jumbo"
    ],
    multidex = "native",
    proguard_specs = [
        "proguard-rules.pro",
    ],
)
```
In the above rule `dexopts` will be ignored

### What operating system are you running Bazel on?

Ubuntu 16.04
MacOS Mojave 10.14.6

### What's the output of `bazel info release`?

0.29.1 and on Master (tag 0.29.1)

### What's the output of `git remote get-url origin ; git rev-parse master ; git rev-parse HEAD` ?

```
$ git remote get-url origin ; git rev-parse master ; git rev-parse HEAD
git@github.sc-corp.net:Snapchat/android.git
5d4f3eabe11aaae99c4714723dcf8c0b1ca67b88
980cb48d5faef727ef592075521894341e37a74a
```

**Comment by ahumesky** [comment]
This probably has something to do with how dx flags are filtered for incremental dexing. Does `dexopts=["--force-jumbo"]` work if you pass `--noincremental_dexing` on the command line?

**Comment by nkoroste** [comment]
Yes, it looks like the dexopts are propagated correctly with `--noincremental_dexing`

**Comment by nkoroste** [comment]
Correction from my previous comment:
- I just noticed that when `--noincremental_dexing` is used my build passes regardless of whether `["--force-jumbo"]` is used or not. Which means I don't know for sure if that flag is propagated successfully or not.

In addition, I just updated to the new 1.2 Bazel version to try the experimental D8 dexer and I noticed the same issue. Where this flag is not propagated and my build fails with incremental dexer:
```
bazel build -s --define=android_incremental_dexing_tool=d8_dexbuilder --nouse_workers_with_dexbuilder //platform/transition/sample:transition-sample
```

But passes fine with `--noincremental_dexing`:
```
$ bazel build -s --noincremental_dexing --define=android_standalone_dexing_tool=d8_compat_dx //platform/transition/sample:transition-sample  
```

That said, from the stack trace I gather that the actual issue is inside DexMerger
```
bazel-out/host/bin/external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexFileMerger
```
Which makes me wonder why do we not use the D8 merger in this case?
